sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/KeyVaultClient.java

```diff
@@ -183,8 +189,21 @@ public static KeyVaultClient createKeyVaultClientBySystemProperty() {
      * @return the access token.
      */
     private String getAccessToken() {
+        if (cacheToken != null && !cacheToken.isExpired()) {
+            return cacheToken.getAccessToken();
+        }
+        cacheToken = getAccessTokenByHttpRequest();
```

Comment:
do we need a lock here?

Reply:
> do we need a lock here?

Hi, KeyVaultClient is not designed for thread-safety use and there is no concurrent use case for it.